airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -224,18 +182,32 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         else:
             return super().get_updated_state(current_stream_state, latest_record)
 
-    def read_records(
+    def stream_slices(
         self,
         sync_mode: SyncMode,
         cursor_field: Optional[List[str]] = None,
-        stream_slice: Optional[Mapping[str, Any]] = None,
         stream_state: Optional[Mapping[str, Any]] = None,
-    ) -> Iterable[StreamData]:
-        yield from super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
-        yield from self.archived_stream.read_records(sync_mode, cursor_field, stream_slice, stream_state)
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        return [{"archived": flag} for flag in (False, True)]
```

Comment:
why does this need a custom component? why can't this be use a ListPartitionRouter?

Reply:
This is still needed for the `CampaignsDetailed` stream, which also requires additional requests to be performed in order to fetch fields which are not available in the main endpoint